transformations/factive_verb_transformation/transformation.py

```diff
@@ -0,0 +1,259 @@
+        return False
+    return True
+
+
+def extract_nsubj_phrase(parse):
+    """
+    extract phrase from nsubj subtree
+    """
+    nsubj_phrase = []
+    for token in parse:
+        if token.dep_ == "nsubj" and token.head.dep_ == "ROOT":
+            nsubj_phrase.append(token.text)
+        if token.head.dep_ == "nsubj" or token.head.head.dep_ == "nsubj":
+            nsubj_phrase.append(token.text)
+    return " ".join(nsubj_phrase)
+
+
+def map_pronoun(word, male_names, female_names):
+    """
+    map word with male and females names, profession, title etc.
+    """
+    pronoun = ""
+    if word in male_names or word.lower() in male_names:
+        pronoun = "he"
+    elif word in female_names or word.lower() in female_names:
+        pronoun = "she"
+    return pronoun
+
+
+def fetch_corresponding_pronoun(nsubj_phrase, male_names, female_names):
+    """
+    Fetch pronoun of nsubj phrase
+    """
+    if nsubj_phrase.lower() in ["i", "you", "we", "he", "she", "they"]:
+        return nsubj_phrase.lower()
+    if len(nsubj_phrase.split(" ")) > 1:
+        for ph in nsubj_phrase.split(" "): # if nsubj phrase contains multiple words.
+            if is_candidate_word(ph):
+                pronoun = map_pronoun(ph, male_names, female_names)
+                if(pronoun != ""):
+                    return pronoun
+        return "they" # default pronoun
+    else:
+        return map_pronoun(nsubj_phrase)
+
+
+def get_transformation(sentence, nlp, factive_verbs, non_factive_verbs, initial_verbs, male_names, female_names, seed):
+    """
+    transform a input sentence by adding factive verb
+    """
+    parse = nlp(sentence)
+    nsubj_phrase = extract_nsubj_phrase(parse)
+    pronoun = fetch_corresponding_pronoun(nsubj_phrase, male_names, female_names)
+    random.seed(0)
+    verb = random.choice(factive_verbs + non_factive_verbs) # pick random verb
+    #initial_verb = random.choice(initial_verbs) # TODO:
+    sentence = sentence.replace(nsubj_phrase, pronoun)
+    return f"{nsubj_phrase} {verb} that, {sentence}"#, f"{nsubj} didn't {verb} that, {sentence}"
+
+
+class FactiveVerbTransformation(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.SENTIMENT_ANALYSIS,
+    ]
+    languages = ["en"]
+    heavy = False
+
+    def __init__(self, seed=1, max_outputs=1):
+        super().__init__(seed, max_outputs=max_outputs)
+        self.nlp = load('en_core_web_sm')
+        self.initial_verbs = ["", "have to", "has to", "need to"] # TODO: use this in next push after discussion
+        #TODO: we can add third person variation like (after discussion)
+        # "Peter published a research paper. => John revealed that, Peter published a research paper."
+        self.male_names, self.female_names = extract_names_and_titles()
+        self.factive_verbs = ["accept", "accepts", "accepted",
+                              "establish", "establishes", "established",
```

Comment:
Also add a Data And Code provenance section mentioning where this data came from.

Reply:
There are some other transformations which use data of male and female and it would be efficient to move those lists to a global directory like spacy and call from there.